File: interviewmaster/invert_binary_tree.py

from collections import deque
from typing import Optional


class TreeNode:
    val: int
   

    def __init__(self, val: int, left: Optional["TreeNode"]=None, right: Optional["TreeNode"] = None):
        self.val = val
        self.left = left
        self.right = right



    def swapNode(self):
        self.left, self.right = self.right, self.left

def  invert_binary_tree(root: TreeNode):

    q = deque([root])
    
    while q:
        removedNode = q.popleft()
        removedNode.swapNode()
        if removedNode.left:
            q.append(removedNode.left)
        if removedNode.right:
            q.append(removedNode.right)
    return root

    

File: javafolder/javainterviewmaster/src/main/java/com/nav/InvertBinaryTree.java
package com.nav;


import java.util.LinkedList;
import java.util.Queue;

public class InvertBinaryTree {

    class TreeNode {

        int val;
        TreeNode left;
        TreeNode right;

        public TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;

        }

        public TreeNode(int val) {
            this.val = val;
        }
    }


    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return root;
        }

        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        while (!q.isEmpty()) {
            TreeNode removedNode = q.poll();
            swapNodes(removedNode);

            if (removedNode.left != null) {
                q.add(removedNode.left);
            }
            if (removedNode.right != null) {
                q.add(removedNode.right);
            }

        }
        return root;
    }


    private void swapNodes(TreeNode current) {
        TreeNode temp = current.left;
        current.left = current.right;
        current.right = temp;

    }
}
